src/pocondest.cc

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2017-2022, University of Tennessee. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+// This program is free software: you can redistribute it and/or modify it under
+// the terms of the BSD 3-Clause license. See the accompanying LICENSE file.
+
+#include "slate/slate.hh"
+#include "auxiliary/Debug.hh"
+#include "slate/Matrix.hh"
+#include "slate/HermitianMatrix.hh"
+#include "internal/internal.hh"
+
+namespace slate {
+
+//------------------------------------------------------------------------------
+/// Distributed parallel estimate of the reciprocal of the condition number
+/// of a general matrix A, in either the 1-norm or the infinity-norm.
+/// Generic implementation for any target.
+///
+/// The reciprocal of the condition number computed as:
+/// \[
+///     rcond = \frac{1}{\|\|A\|\| \times \|\|A^{-1}\|\-}
+/// \]
+/// where $A$ is the output of the Cholesky factorization (potrf).
+///
+//------------------------------------------------------------------------------
+/// @tparam scalar_t
+///     One of float, double, std::complex<float>, std::complex<double>.
+//------------------------------------------------------------------------------
+/// @param[in] in_norm
+///     rcond to compute:
+///     - Norm::One: 1-norm condition number
+///     - Norm::Inf: infinity-norm condition number
+///
+/// @param[in] A
+///     On entry, the n-by-n matrix $A$.
+///     It is the output of the Cholesky factorization of a Hermitian matrix.
+///
+/// @param[in] Anorm
+///     If Norm::One, the 1-norm of the original matrix A.
+///     If Norm::Inf, the infinity-norm of the original matrix A.
+///
+/// @param[in,out] rcond
+///     The reciprocal of the condition number of the matrix A,
+///     computed as stated above.
+///
+/// @param[in] opts
+///     Additional options, as map of name = value pairs. Possible options:
+///     - Option::Target:
+///       Implementation to target. Possible values:
+///       - HostTask:  OpenMP tasks on CPU host [default].
+///       - HostNest:  nested OpenMP parallel for loop on CPU host.
+///       - HostBatch: batched BLAS on CPU host.
+///       - Devices:   batched BLAS on GPU device.
+///
+/// @ingroup cond_specialization
+///
+template <typename scalar_t>
+void pocondest(
+           Norm in_norm,
+           HermitianMatrix<scalar_t>& A,
+           blas::real_type<scalar_t> *Anorm,
```

Comment:
Is there any reason `Anorm` is a pointer? Also in `gecondest`.

Reply:
I initially just followed gecondest.  I had intended to go back and replace it with pass-by-value, but I forgot.  So, I just made that change.